DataFormats/Portable/interface/PortableCollectionCommon.h

```diff
@@ -0,0 +1,113 @@
+
+#ifndef DataFormats_Portable_interface_PortableCollectionCommon_h
+#define DataFormats_Portable_interface_PortableCollectionCommon_h
+
+template <auto Start, auto End, auto Inc, class F>
+constexpr void constexpr_for(F&& f) {
+  if constexpr (Start < End) {
+    f(std::integral_constant<decltype(Start), Start>());
+    constexpr_for<Start + Inc, End, Inc>(f);
+  }
+}
+
+template <std::size_t idx, typename T>
+struct CollectionLeaf {
+  CollectionLeaf() = default;
+  CollectionLeaf(std::byte* buffer, int32_t elements) : layout_(buffer, elements), view_(layout_) {}
```

Comment:
I would wither change this to
```suggestion
  CollectionLeaf(std::byte* buffer, int32_t size) : layout_(buffer, size), view_(layout_) {}
```
or use `elements` instead of `sizes` in the next one